kratos/solving_strategies/builder_and_solvers/builder_and_solver.h

```diff
@@ -206,9 +205,29 @@ class BuilderAndSolver
      */
     /*@{ */
 
+    /**
+     * @brief Default constructor. (with parameters)
+     */
+    explicit BuilderAndSolver(typename TLinearSolver::Pointer pNewLinearSystemSolver, Parameters ThisParameters)
+    {
+        // Validate default parameters
+        Parameters default_parameters = Parameters(R"(
+        {
+        })" );
+
+        ThisParameters.ValidateAndAssignDefaults(default_parameters);
+
+        // We set the other member variables
+        mpLinearSystemSolver = pNewLinearSystemSolver;
+
+        mDofSetIsInitialized = false;
+
+        mReshapeMatrixFlag = false;
+    }
+
     /** Constructor.
      */
-    BuilderAndSolver(
+    explicit BuilderAndSolver(
         typename TLinearSolver::Pointer pNewLinearSystemSolver)
     {
         mpLinearSystemSolver = pNewLinearSystemSolver;
```

Comment:
not essential, but you could call here the constructor by parameters

Reply:
For the linear solver?, no, as I said the Linear solver is already constructed in the Strategy, if you do that you will create two linear solvers, which is not very consistent

Reply:
you could have done
 
     explicit BuilderAndSolver(
        typename TLinearSolver::Pointer pNewLinearSystemSolver)
     {
            BuilderAndSolver(pNewLinearSystemSolver, Parameters() )
     }

and essentially implement one single constructor

OR EVEN just define one single constructor passing by default an empty parametes object

Reply:
Huuum, at the end is the same thing. Can be changed in the future